File: src/main/java/net/jgp/books/spark/ch07/lab920_orc_ingestion/OrcToDataframeApp.java
package net.jgp.books.spark.ch07.lab920_orc_ingestion;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

/**
 * ORC ingestion in a dataframe.
 * 
 * Source of file: Apache ORC project -
 * https://github.com/apache/orc/tree/master/examples
 * 
 * @author jgp
 */
public class OrcToDataframeApp {

  /**
   * main() is your entry point to the application.
   * 
   * @param args
   */
  public static void main(String[] args) {
    OrcToDataframeApp app = new OrcToDataframeApp();
    app.start();
  }

  /**
   * The processing code.
   */
  private void start() {
    // Creates a session on a local master
    SparkSession spark = SparkSession.builder()
        .appName("ORC to Dataframe")
        .config("spark.sql.orc.impl", "native")
        .master("local")
        .getOrCreate();

    // Reads an ORC file, stores it in a dataframe
    Dataset<Row> df = spark.read()
        .format("orc")
        .load("data/demo-11-zlib.orc");

    // Shows at most 10 rows from the dataframe
    df.show(10);
    df.printSchema();
    System.out.println("The dataframe has " + df.count() + " rows.");
  }
}

File: src/main/python/lab920_orc_ingestion/orcToDataframeApp.py
"""
 ORC ingestion in a dataframe.

 Source of file: Apache ORC project -
 https://github.com/apache/orc/tree/master/examples

 @author rambabu.posa
"""
from pyspark.sql import SparkSession
import os

current_dir = os.path.dirname(__file__)
relative_path = "../../../../data/demo-11-zlib.orc"
absolute_file_path = os.path.join(current_dir, relative_path)

# Creates a session on a local master
spark = SparkSession.builder.appName("ORC to Dataframe") \
    .config("spark.sql.orc.impl", "native") \
    .master("local[*]").getOrCreate()

# Reads an ORC file, stores it in a dataframe
df = spark.read.format("orc") \
    .load(absolute_file_path)

# Shows at most 10 rows from the dataframe
df.show(10)
df.printSchema()

print("The dataframe has {} rows.".format(df.count()))

spark.stop()